src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,258 @@
+        String lastInput = null;
+
+        while (!this.isExited) {
+            System.out.println("");
+            lastInput = currScanner.nextLine();
+
+            try {
+                this.processCmdInput(lastInput);
+            } catch (DukeExceptionBase dukeE) {
+                dukeE.dukeSayErrorMsg();
+            }
+        }
+
+
+    }
+
+    /**
+     * Runs to process the command input using the command parser.
+     *
+     * @param lastInput   The command to process.
+     * @throws DukeExceptionBase when an invalid command gets entered.
+     */
+    private void processCmdInput(String lastInput) throws DukeExceptionBase {
+        // Get the type of command that this input represents
+        DukeCommandParser.CommandType cmdType = this.currDukeCmdParser.parse(lastInput);
+
+        BaseTask.TaskType currTaskType = BaseTask.checkTaskType(lastInput);
+
+
+        if (cmdType == DukeCommandParser.CommandType.BYE) {
+            this.dukeExiter();
+        } else if (cmdType == DukeCommandParser.CommandType.LIST) {
+            this.listOutTDL();
+        } else if (cmdType == DukeCommandParser.CommandType.MARK_TASK_DONE) {
+            this.markItemDoneInTDL(lastInput);
+        } else if (cmdType == DukeCommandParser.CommandType.DEL_TASK) {
+            this.deleteTaskInTDL(lastInput);
+        } else if (cmdType == DukeCommandParser.CommandType.ADD_TASK) {
+            this.addToTDL(lastInput, currTaskType);
+        } else if (cmdType == DukeCommandParser.CommandType.FIND) {
+            this.findTaskInTDL(lastInput);
+
+        } else {
+            unknownCommandEntered();
+        }
+    }
+
+    /**
+     * Runs when program is going to exit.
+     */
+    private void dukeExiter() {
+        dukeSays("Bye. Hope to see you soon!");
+        this.isExited = true;
+    }
+
+    /**
+     * Used when Duke is supposed to say something.
+     * This would use the UI controller to print the message between 2 line separators.
+     *
+     * @param printThis The message to print inside Duke's text bubble
+     */
+    public static void dukeSays(String printThis) {
+        Duke currDuke = Duke.getCurrDuke();
+
+        // Updated to work with new UI Controller which contains a buffer
+        currDuke.currUiCtrl.addToDukeBuffer(printThis);
+        currDuke.currUiCtrl.dukeBufferRelease();
+    }
+
+    /**
+     * Used to add something that Duke will say later into it's UI buffer.
+     *
+     * @param printLater The message that Duke will say when the buffer is released.
+     */
+    public static void dukeLaterSay(String printLater) {
+        Duke currDuke = Duke.getCurrDuke();
+
+        currDuke.currUiCtrl.addToDukeBuffer(printLater);
+    }
+
+    private void unknownCommandEntered() throws DukeExceptionBase {
+        throw new DukeExceptionBase("Please enter something valid!");
+    }
+
+    private void addToTDL(String str, BaseTask.TaskType currTaskType) throws DukeExceptionBase {
```

Comment:
Should it be addToTdl instead? Reference: Abbreviations and acronyms should not be uppercase when used as a part of a name.